pygmt/src/basemap.py

```diff
@@ -87,6 +91,8 @@ def basemap(self, **kwargs):
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
     if not args_in_kwargs(args=["B", "L", "Td", "Tm", "c"], kwargs=kwargs):
-        kwargs["B"] = True  # Plotting frames if required arguments not given
+        raise GMTInvalidInput(
+            "At least one of 'frame', 'map_scale', 'compass', 'rose', or 'panel' must be specified."
+        )
```

Comment:
Actually, in subplot mode, none of these options are required. For example, the first example in https://github.com/GenericMappingTools/pygmt/issues/2365 no longer works after this PR.

Ideally, we should check if it's in subplot mode, and these options are required only when it's NOT in subplot mode. However, currently we can't know it. So I believe we have to remove the checks, right?

Reply:
Done in commit https://github.com/GenericMappingTools/pygmt/pull/2417/commits/8e9d4e52cdccba20d835b86cd2f11e0e20d24c7b.